src/app_logic.py

```diff
@@ -11,11 +11,13 @@
 NODE_PREFIX = "node_"
 
 class CounterApp(Thread):
-    def __init__(self):
+    def __init__(self, leader_manager):
         Thread.__init__(self)
```

Comment:
Not sure it's related to your PR but why do we need `Thread` here?

Reply:
Probably not related to this PR but I think it is there so that we can run the CounterApp.run() in a separate thread. It has to be in a separate thread because app.run() is blocking.